src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/Metadata/ModelMetadataIdentity.cs

```diff
@@ -66,17 +66,37 @@ public static ModelMetadataIdentity ForType(Type modelType)
             };
         }
 
+        /// <summary>
+        /// Creates a <see cref="ModelMetadataIdentity"/> for the provided parameter.
+        /// </summary>
+        /// <param name="parameter">The <see cref="ParameterInfo" />.</param>
+        /// <returns>A <see cref="ModelMetadataIdentity"/>.</returns>
         public static ModelMetadataIdentity ForParameter(ParameterInfo parameter)
+            => ForParameter(parameter, parameter?.ParameterType);
+
+        /// <summary>
+        /// Creates a <see cref="ModelMetadataIdentity"/> for the provided parameter with the specified
+        /// model type.
+        /// </summary>
+        /// <param name="parameter">The <see cref="ParameterInfo" />.</param>
+        /// <param name="modelType">The model type.</param>
+        /// <returns>A <see cref="ModelMetadataIdentity"/>.</returns>
+        private static ModelMetadataIdentity ForParameter(ParameterInfo parameter, Type modelType)
         {
             if (parameter == null)
             {
                 throw new ArgumentNullException(nameof(parameter));
             }
 
+            if (modelType == null)
+            {
+                throw new ArgumentNullException(nameof(modelType));
```

Comment:
So we're enforcing the modelType now? This used to not fail in the original code.

Reply:
@mkArtakMSFT this can't fail in the new code. It'll get `ParameterInfo.ModelType` when called from [line 75](https://github.com/aspnet/Mvc/pull/8049/files#diff-c6ca47a121360cde41735d61897a12acR75) or the (non-`null`) runtime type of the parameter when called using Reflection.

Reply:
So the expectation is that the ParameterType will never be null for a non-null parameter

Reply:
More specifically, this method's `modelType` parameter will never be `null` as we use it. The `null` check exists because this method is `public` in 2.2.